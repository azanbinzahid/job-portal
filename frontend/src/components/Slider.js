import React from 'react'
import { Carousel } from 'react-bootstrap'
import 'holderjs'

export default function Slider() {
    return (
        <Carousel>
        <Carousel.Item>
            <img
            className="d-block w-100"
            src="/images/1.jpg"
            alt="First slide"
            />
        </Carousel.Item>
        <Carousel.Item>
            <img
            className="d-block w-100"
            src="/images/2.jpg"
            alt="First slide"
            />
        </Carousel.Item>

        </Carousel>    
        )
}
