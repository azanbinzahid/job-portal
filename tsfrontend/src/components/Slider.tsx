import React, { FC } from "react";
import { Carousel } from "react-bootstrap";

const Slider: FC = () => {
  return (
    <Carousel>
      <Carousel.Item>
        <img className="d-block w-100" src="/images/1.jpg" alt="First slide" />
      </Carousel.Item>
      <Carousel.Item>
        <img className="d-block w-100" src="/images/2.jpg" alt="First slide" />
      </Carousel.Item>
    </Carousel>
  );
};

export default Slider;
