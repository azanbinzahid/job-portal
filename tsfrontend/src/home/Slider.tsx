import Search from "jobs/Search";
import React, { FC } from "react";
import { Carousel } from "react-bootstrap";
import styled from "styled-components";

const BannerTitle = styled.h1`
  display: block;
  margin-bottom: 12px;
  font-size: 47px;
  font-family: "Jost", "PT Serif", sans-serif;
  font-weight: 700;
  color: #fff;
`;

const BannerText = styled.p`
  display: block;
  margin-bottom: 12px;
  font-family: "Jost", "PT Serif", sans-serif;
  font-weight: 700;
  color: #fff;
`;

const Image = styled.img`
  filter: brightness(50%);
`;

const Slider: FC = () => {
  return (
    <Carousel>
      <Carousel.Item>
        <Image
          // className="d-block w-100"
          src="https://www.jobs.punjab.gov.pk/new_design/images/banner2.jpg"
          alt="First slide"
        />
        <Carousel.Caption>
          <BannerTitle>Get The Career You Deserve</BannerTitle>
          <Search />
          <BannerText> For example: remote, lahore, manager </BannerText>
        </Carousel.Caption>
      </Carousel.Item>
    </Carousel>
  );
};

export default Slider;
