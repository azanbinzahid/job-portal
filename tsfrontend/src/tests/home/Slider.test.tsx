import React from "react";
import { shallow } from "enzyme";
import Slider from "home/Slider";
test("renders the component", () => {
  const component = shallow(<Slider />);
  expect(component).toMatchSnapshot();
});
