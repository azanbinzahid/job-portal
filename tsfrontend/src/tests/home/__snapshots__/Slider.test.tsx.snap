// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <Carousel
    className="d-none d-xl-block"
    controls={true}
    defaultActiveIndex={0}
    fade={false}
    indicators={true}
    interval={5000}
    keyboard={true}
    nextIcon={<React.Fragment />}
    nextLabel="Next"
    pause="hover"
    prevIcon={<React.Fragment />}
    prevLabel="Previous"
    slide={true}
    touch={true}
    wrap={true}
  >
    <CarouselItem>
      <styled.img
        alt="First slide"
        className="d-block w-100 mh-50"
        src="https://www.jobs.punjab.gov.pk/new_design/images/banner2.jpg"
      />
      <CarouselCaption>
        <div>
          <img
            alt=""
            className="d-inline-block align-top pb-3"
            height="150"
            src="https://image.flaticon.com/icons/svg/3232/3232975.svg"
            width="150"
          />
          <styled.h1>
            <span>
               Get The Career You Deserve 
            </span>
          </styled.h1>
          <withRouter(Search) />
          <styled.p>
            <span>
              Example keywords: remote, lahore, manager 
            </span>
          </styled.p>
        </div>
      </CarouselCaption>
    </CarouselItem>
  </Carousel>
  <div
    className="d-xl-none pt-5"
  >
    <SectionHeading
      subtitle="Get the career you deserve"
      title="Serach for jobs"
    />
    <withRouter(Search) />
  </div>
</Fragment>
`;
